CMake/ClangTidy.cmake

```diff
@@ -0,0 +1,31 @@
+if (clang_tidy)
+
+  function (find_clang_tidy _CLANG_TIDY_PATH)
+    if (DEFINED ENV{CLIO_CLANG_TIDY_BIN})
+      set (_CLANG_TIDY_PATH $ENV{CLIO_CLANG_TIDY_BIN})
+      if (NOT EXISTS ${_CLANG_TIDY_PATH})
+        message (FATAL_ERROR "$ENV{CLIO_CLANG_TIDY_BIN} no such file. Check CLIO_CLANG_TIDY_BIN env variable")
+      endif ()
+      message (STATUS "Using clang-tidy from CLIO_CLANG_TIDY_BIN")
+    else ()
+      find_program (_CLANG_TIDY_PATH NAMES "clang-tidy-16" "clang-tidy-15" "clang-tidy-14" "clang-tidy" REQUIRED)
```

Comment:
find_program is supposed to create the var to carry the result. not sure if existing variable can reflect the result correctly? 
I wonder if you change this line to "      find_program (_CLANG_TIDY_PATH NAMES "bb" "cc" "dd" "ee" REQUIRED)
",  if (NOT _CLANG_TIDY_PATH) still is false

Reply:
The below snippet is to try to carry the update from the argument, but it fails for me 
```  function (mmm TMPMM)
  set(TMPMM, "123")
  endfunction ()

  mmm(TMPMM)  
  message(print "${TMPMM}")
```
I might misunderstand this part.

Reply:
Now it should be better